### PR TITLE
feature: New module daikon-spring-security

### PR DESCRIPTION
**Pull request by fhuaulme**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the new behavior?**
See README for details 

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:

Provides helper classes relative to Spring Security

[@RequiresAuthority]
* Add annotation for permission check with custom onDeny behavior.
* Move code from Data Prep to Daikon.

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/22/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/22/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/23/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/23/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/39/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/39/)

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        LOGGER.debug("Checking @Required access on {} for user {}.", pjp, authentication);
+
+        final MethodSignature methodSignature = (MethodSignature) pjp.getSignature();
+        final Method method = methodSignature.getMethod();
+
+        final RequiresAuthority annotation = method.getAnnotation(RequiresAuthority.class);
+        if (annotation == null) {
+            throw new IllegalArgumentException("Missing @RequiresAuthority annotation."); // Rather unexpected
+        }
+        final String authority = of(annotation.authority()) //
```
Comment (sgandon) [inline daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java]: Just for my understanding what is the difference with
`final String authority = (annotation.authority() == null ? annotation.value() : annotation.authority());`

Comment (fhuaulme) [inline daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/access/RequiresAuthorityAspect.java]: No difference at all in fact.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/64/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/64/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/65/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/65/)

File: daikon-spring/daikon-spring-security/pom.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            <version>1.2.5.RELEASE</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.aspectj</groupId>
+            <artifactId>aspectjweaver</artifactId>
+            <version>1.8.9</version>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
```
Comment (fhuaulme) [inline daikon-spring/daikon-spring-security/pom.xml]: Done, should have used slf4j-api iso. logback (wrong usage of dependency search, my bad).

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/66/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/66/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/67/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/67/)

**Pull request merged**